<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Visualising taxonomy with KRONA &mdash; Metagenomics Workshop SciLifeLab 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Metagenomics Workshop SciLifeLab 1.0 documentation" href="../index.html" />
    <link rel="up" title="Comparative Taxonomic Analysis of Metagenomic Samples" href="index.html" />
    <link rel="next" title="Comparative taxonomic analysis with R" href="compare.html" />
    <link rel="prev" title="Extracting rRNA encoding reads and annotating them" href="rrna.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="compare.html" title="Comparative taxonomic analysis with R"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="rrna.html" title="Extracting rRNA encoding reads and annotating them"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" accesskey="U">Comparative Taxonomic Analysis of Metagenomic Samples</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="visualising-taxonomy-with-krona">
<h1>Visualising taxonomy with KRONA<a class="headerlink" href="#visualising-taxonomy-with-krona" title="Permalink to this headline">Â¶</a></h1>
<p>To get a graphical representation of the taxonomic classifications you can use
KRONA, which is an excellent program for exploring data with hierarchical
structures in general. The output file is an html file that can be viewed in a
browser. Again make a directory for KRONA:</p>
<div class="highlight-python"><pre>mkdir -p ~/metagenomics_workshop2/krona
cd ~/metagenomics_workshop2/krona</pre>
</div>
<p>And run KRONA, concatenating the archaea and bacteria class files together at the same time like this and providing the name of the sample:</p>
<div class="highlight-python"><pre>ktImportRDP \
&lt;(cat ../rdp/0328_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/0328_rrna.silva-bac-16s-database-id85.fasta.class.tsv),0328 \
&lt;(cat ../rdp/0403_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/0403_rrna.silva-bac-16s-database-id85.fasta.class.tsv),0403 \
&lt;(cat ../rdp/0423_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/0423_rrna.silva-bac-16s-database-id85.fasta.class.tsv),0423 \
&lt;(cat ../rdp/0531_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/0531_rrna.silva-bac-16s-database-id85.fasta.class.tsv),0531 \
&lt;(cat ../rdp/0619_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/0619_rrna.silva-bac-16s-database-id85.fasta.class.tsv),0619 \
&lt;(cat ../rdp/0705_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/0705_rrna.silva-bac-16s-database-id85.fasta.class.tsv),0705 \
&lt;(cat ../rdp/0709_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/0709_rrna.silva-bac-16s-database-id85.fasta.class.tsv),0709 \
&lt;(cat ../rdp/1001_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/1001_rrna.silva-bac-16s-database-id85.fasta.class.tsv),1001 \
&lt;(cat ../rdp/1004_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/1004_rrna.silva-bac-16s-database-id85.fasta.class.tsv),1004 \
&lt;(cat ../rdp/1028_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/1028_rrna.silva-bac-16s-database-id85.fasta.class.tsv),1028 \
&lt;(cat ../rdp/1123_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/1123_rrna.silva-bac-16s-database-id85.fasta.class.tsv),1123</pre>
</div>
<p>The <tt class="docutils literal"><span class="pre">&lt;()</span></tt> in bash can be used for process substitution
(<a class="reference external" href="http://tldp.org/LDP/abs/html/process-sub.html">http://tldp.org/LDP/abs/html/process-sub.html</a> ). Just for your information,
the above command was actually generated with the following commands:</p>
<div class="highlight-python"><pre>cmd=`echo ktImportRDP; for s in ${samplenames[*]}; do echo '&lt;('cat ../rdp/${s}_rrna.silva-arc-16s-database-id95.fasta.class.tsv ../rdp/${s}_rrna.silva-bac-16s-database-id85.fasta.class.tsv')',$s; done`
echo $cmd</pre>
</div>
<p>Copy the resulting file rdp.krona.html to your local computer with scp and open it in firefox.</p>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="rrna.html"
                        title="previous chapter">Extracting rRNA encoding reads and annotating them</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="compare.html"
                        title="next chapter">Comparative taxonomic analysis with R</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../_sources/comparative-taxonomic-analysis/krona.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="compare.html" title="Comparative taxonomic analysis with R"
             >next</a> |</li>
        <li class="right" >
          <a href="rrna.html" title="Extracting rRNA encoding reads and annotating them"
             >previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" >Comparative Taxonomic Analysis of Metagenomic Samples</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Johannes Alneberg, Johan Bengtsson-Palme, Ino de Bruijn, Luisa Hugerth, Mikael Huss, Thomas Svensson.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b3.
    </div>
  </body>
</html>