<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Extracting rRNA encoding reads and annotating them &mdash; Metagenomics Workshop SciLifeLab 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Metagenomics Workshop SciLifeLab 1.0 documentation" href="../index.html" />
    <link rel="up" title="Comparative Taxonomic Analysis Workshop" href="index.html" />
    <link rel="next" title="Visualising taxonomy with KRONA" href="krona.html" />
    <link rel="prev" title="Comparative Taxonomic Analysis Workshop" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="krona.html" title="Visualising taxonomy with KRONA"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Comparative Taxonomic Analysis Workshop"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" accesskey="U">Comparative Taxonomic Analysis Workshop</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="extracting-rrna-encoding-reads-and-annotating-them">
<h1>Extracting rRNA encoding reads and annotating them<a class="headerlink" href="#extracting-rrna-encoding-reads-and-annotating-them" title="Permalink to this headline">¶</a></h1>
<p>Taxonomic composition of a sample can be based on e.g. BLASTing the contigs
against a database of reference genomes, like you will learn how to do
tomorrow, or by utilising rRNA sequences. Usually assembly doesn’t work well
for rRNA genes due to their highly conserved regions, therefore extracting rRNA
from contigs will miss a lot of the taxonomic information that can be obtained
by analysing the reads directly. Analysing the reads also has the advantage of
being quantitative, i.e. we don’t need to calculate coverages by the mapping
procedure we applied for the functional genes above. We will extract rRNA
encoding reads with the program sortmeRNA which is one of the fastest softwares
for this (the program rnammer that you will use tomorrow is not good for
partial rRNA sequences as those on short reads). The program sortmeRNA has
built-in multithreading support so this time we use that for parallelization
instead of gnu parallel. These are the commands to run:</p>
<div class="highlight-python"><div class="highlight"><pre>mkdir -p ~/metagenomics_workshop2/sortmerna
cd ~/metagenomics_workshop2/sortmerna
for s in ${samplenames[*]}
    do sortmerna -n 2 --db \
    /glob/inod/src/sortmerna-1.9/rRNA_databases/silva-arc-16s-database-id95.fasta \
    /glob/inod/src/sortmerna-1.9/rRNA_databases/silva-bac-16s-database-id85.fasta \
    --I /proj/g2013206/metagenomics/reads/${s}_pe.fasta \
    --accept ${s}_rrna \
    --other ${s}_nonrrna  \
    --bydbs  \
    -a 8  \
    --log ${s}_bilan  \
    -m 5242880
done
</pre></div>
</div>
<p>Again, this command takes rather long to run (~5m per sample) so just copy the results if you don’t feel like waiting:</p>
<div class="highlight-python"><div class="highlight"><pre>cp /proj/g2013206/metagenomics/sortmerna/* ~/metagenomics_workshop2/sortmerna
</pre></div>
</div>
<p>It outputs the reads or part of reads that encode rRNA in a fasta file. These rRNA sequences can be classified in many ways, and again blasting them against a suitable database is one option. Here we use a simple and fast method (unless you have too many samples), the classifier tool at RDP (ribosomal database project). This uses a naive bayesian classifier trained on many sequences of defined taxonomies. It gives bootstrap support values for each taxonomic level; usually the support gets lower the further down the hierarchy you go. Genus level is the lowest level provided. You can use the web service if you prefer, and upload each file individually, or you can use the uppmax installation of RDP classifier like this (~4m):</p>
<div class="highlight-python"><div class="highlight"><pre>mkdir -p ~/metagenomics_workshop2/rdp
cd ~/metagenomics_workshop2/rdp
for s in ../sortmerna/*_rrna*.fasta
    do java -Xmx1g -jar /glob/inod/src/rdp_classifier_2.6/dist/classifier.jar \
    classify \
    -g 16srrna  \
    -b `basename ${s}`.bootstrap  \
    -h `basename ${s}`.hier.tsv  \
    -o `basename ${s}`.class.tsv  \
    ${s}
done
</pre></div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">Comparative Taxonomic Analysis Workshop</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="krona.html"
                        title="next chapter">Visualising taxonomy with KRONA</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../_sources/comparative-taxonomic-analysis/rrna.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="krona.html" title="Visualising taxonomy with KRONA"
             >next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Comparative Taxonomic Analysis Workshop"
             >previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" >Comparative Taxonomic Analysis Workshop</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Johannes Alneberg, Johan Bengtsson-Palme, Ino de Bruijn, Luisa Hugerth, Mikael Huss, Thomas Svensson.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.2.
    </div>
  </body>
</html>