<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Mapping reads back to the assembly &mdash; Metagenomics Workshop SciLifeLab 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Metagenomics Workshop SciLifeLab 1.0 documentation" href="../index.html" />
    <link rel="up" title="Metagenomic Assembly Workshop" href="index.html" />
    <link rel="next" title="Comparative Functional Analysis Workshop" href="../comparative-functional-analysis/index.html" />
    <link rel="prev" title="Assembling reads with Velvet" href="assembly.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../comparative-functional-analysis/index.html" title="Comparative Functional Analysis Workshop"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="assembly.html" title="Assembling reads with Velvet"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" accesskey="U">Metagenomic Assembly Workshop</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="mapping-reads-back-to-the-assembly">
<h1>Mapping reads back to the assembly<a class="headerlink" href="#mapping-reads-back-to-the-assembly" title="Permalink to this headline">¶</a></h1>
<div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>There are many different mappers available to map your reads back to the
assemblies. Usually they result in a SAM or BAM file
(<a class="reference external" href="http://genome.sph.umich.edu/wiki/SAM">http://genome.sph.umich.edu/wiki/SAM</a>). Those are formats that contain the
alignment information, where BAM is the binary version of the plain text SAM
format. In this tutorial we will be using bowtie2
(<a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a>).</p>
<p>The SAM/BAM file can afterwards be processed with Picard
(<a class="reference external" href="http://picard.sourceforge.net/">http://picard.sourceforge.net/</a>) to remove duplicate reads. Those are likely to
be reads that come from a PCR duplicate (<a class="reference external" href="http://www.biostars.org/p/15818/">http://www.biostars.org/p/15818/</a>).</p>
<p>BEDTools (<a class="reference external" href="http://code.google.com/p/bedtools/">http://code.google.com/p/bedtools/</a>) can then be used to retrieve
coverage statistics.</p>
<p>There is a script available that does it all at once. Read it and try to
understand what happens in each step:</p>
<div class="highlight-python"><div class="highlight"><pre>less `which map-bowtie2-markduplicates.sh`
map-bowtie2-markduplicates.sh -h
</pre></div>
</div>
<p>Bowtie2 has some nice documentation: <a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml</a></p>
<p><strong>Question: what does bowtie2-build do?</strong></p>
<p>Picard&#8217;s documentation also exists! Two bioinformatics programs in a row with
decent documentation! Take a moment to celebrate, then have a look here:
<a class="reference external" href="http://sourceforge.net/apps/mediawiki/picard/index.php?title=Main_Pageon">http://sourceforge.net/apps/mediawiki/picard/index.php?title=Main_Pageon</a></p>
<p><strong>Question: Why not just remove all identitical pairs instead of mapping them
and then removing them?</strong></p>
<p><strong>Question: What is the difference between samtools rmdup and Picard MarkDuplicates?</strong></p>
</div>
<div class="section" id="mapping-reads-with-bowtie2">
<h2>Mapping reads with bowtie2<a class="headerlink" href="#mapping-reads-with-bowtie2" title="Permalink to this headline">¶</a></h2>
<p>Take an assembly and try to map the reads back using bowtie2. Do this on an
interactive node again, and remember to change the &#8216;out_21&#8217; part to the actual output directory that you generated:</p>
<div class="highlight-python"><div class="highlight"><pre># Create a new directory and link files
mkdir -p ~/asm-workshop/bowtie2
cd ~/asm-workshop/bowtie2
ln -s ../velvet/out_21/contigs.fa contigs.fa
ln -s ../sickle/pair1.fastq pair1.fastq
ln -s ../sickle/pair2.fastq pair2.fastq

# Run the everything in one go script.
map-bowtie2-markduplicates.sh -t 1 -c pair1.fastq pair2.fastq pair contigs.fa contigs map &gt; map.log
</pre></div>
</div>
<p>Inspect the <tt class="docutils literal"><span class="pre">map.log</span></tt> output and see if all went well.</p>
<p><strong>Question: What is the overall alignment rate of your reads that bowtie2 reports?</strong></p>
<p>Add the answer to the <a class="reference external" href="https://docs.google.com/spreadsheet/ccc?key=0AvduvUOYAB-_dDdDSVhqUi1KQmJkTlZJcHVfMGI3a2c#gid=3">doc</a>.</p>
</div>
<div class="section" id="some-general-statistics-from-the-sam-bam-file">
<h2>Some general statistics from the SAM/BAM file<a class="headerlink" href="#some-general-statistics-from-the-sam-bam-file" title="Permalink to this headline">¶</a></h2>
<p>You can also determine mapping statistics directly from the bam file. Use for
instance:</p>
<div class="highlight-python"><div class="highlight"><pre># Mapped reads only
samtools view -c -F 4 map/contigs_pair-smds.bam

# Unmapped reads only
samtools view -c -f 4 map/contigs_pair-smds.bam
</pre></div>
</div>
<p>From:
<a class="reference external" href="http://left.subtree.org/2012/04/13/counting-the-number-of-reads-in-a-bam-file/">http://left.subtree.org/2012/04/13/counting-the-number-of-reads-in-a-bam-file/</a>.
The number is different from the number that bowtie2 reports, because these are
the numbers after removing duplicates. The <tt class="docutils literal"><span class="pre">-smds</span></tt> part stands for running
<tt class="docutils literal"><span class="pre">samtools</span> <span class="pre">sort</span></tt>, <tt class="docutils literal"><span class="pre">MarkDuplicates.jar</span></tt> and <tt class="docutils literal"><span class="pre">samtools</span> <span class="pre">sort</span></tt> again on the
bam file. If all went well with the mapping there should also be a
<tt class="docutils literal"><span class="pre">map/contigs_pair-smd.metrics</span></tt> file where you can see the percentage of
duplication. Add that to the <a class="reference external" href="https://docs.google.com/spreadsheet/ccc?key=0AvduvUOYAB-_dDdDSVhqUi1KQmJkTlZJcHVfMGI3a2c#gid=3">doc</a> as well.</p>
</div>
<div class="section" id="coverage-information-from-bedtools">
<h2>Coverage information from BEDTools<a class="headerlink" href="#coverage-information-from-bedtools" title="Permalink to this headline">¶</a></h2>
<p>Look at the output from BEDTools:</p>
<div class="highlight-python"><div class="highlight"><pre>less map/contigs_pair-smds.coverage
</pre></div>
</div>
<p>The format is explained here
<a class="reference external" href="http://bedtools.readthedocs.org/en/latest/content/tools/genomecov.html">http://bedtools.readthedocs.org/en/latest/content/tools/genomecov.html</a>. The
<tt class="docutils literal"><span class="pre">map-bowtie2-markduplicates.sh</span></tt> script also outputs the mean coverage per
contig:</p>
<div class="highlight-python"><div class="highlight"><pre>less map/contigs_pair-smds.coverage.percontig
</pre></div>
</div>
<p><strong>Question: What is the contig with the highest coverage? Hint: use sort -k</strong></p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Mapping reads back to the assembly</a><ul>
<li><a class="reference internal" href="#overview">Overview</a></li>
<li><a class="reference internal" href="#mapping-reads-with-bowtie2">Mapping reads with bowtie2</a></li>
<li><a class="reference internal" href="#some-general-statistics-from-the-sam-bam-file">Some general statistics from the SAM/BAM file</a></li>
<li><a class="reference internal" href="#coverage-information-from-bedtools">Coverage information from BEDTools</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="assembly.html"
                        title="previous chapter">Assembling reads with Velvet</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="../comparative-functional-analysis/index.html"
                        title="next chapter">Comparative Functional Analysis Workshop</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../_sources/assembly/map.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../comparative-functional-analysis/index.html" title="Comparative Functional Analysis Workshop"
             >next</a> |</li>
        <li class="right" >
          <a href="assembly.html" title="Assembling reads with Velvet"
             >previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" >Metagenomic Assembly Workshop</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Johannes Alneberg, Johan Bengtsson-Palme, Ino de Bruijn, Luisa Hugerth, Mikael Huss, Thomas Svensson.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.2.
    </div>
  </body>
</html>