<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Checking required software &mdash; Metagenomics Workshop SciLifeLab 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Metagenomics Workshop SciLifeLab 1.0 documentation" href="../index.html" />
    <link rel="up" title="Metagenomic Assembly Workshop" href="index.html" />
    <link rel="next" title="Quality trimming Illumina paired-end reads" href="qtrim.html" />
    <link rel="prev" title="Metagenomic Assembly Workshop" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="qtrim.html" title="Quality trimming Illumina paired-end reads"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Metagenomic Assembly Workshop"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" accesskey="U">Metagenomic Assembly Workshop</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="checking-required-software">
<h1>Checking required software<a class="headerlink" href="#checking-required-software" title="Permalink to this headline">¶</a></h1>
<p>An often occuring theme in bioinformatics is installing software. Here we wil
go over some steps to help you check whether you actually have the right
software installed. There&#8217;s an optional excerise on how to install <tt class="docutils literal"><span class="pre">sickle</span></tt>.</p>
<div class="section" id="programs-used-in-this-workshop">
<h2>Programs used in this workshop<a class="headerlink" href="#programs-used-in-this-workshop" title="Permalink to this headline">¶</a></h2>
<p>The following programs are used in this workshop:</p>
<blockquote>
<div><ul class="simple">
<li><a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a></li>
<li><a class="reference external" href="http://www.ebi.ac.uk/~zerbino/velvet/">Velvet</a></li>
<li><a class="reference external" href="https://www.gnu.org/software/parallel/">parallel</a></li>
<li><a class="reference external" href="http://samtools.sourceforge.net/">samtools</a></li>
<li><a class="reference external" href="https://code.google.com/p/cdhit/">CD-HIT</a></li>
<li><a class="reference external" href="http://sourceforge.net/apps/mediawiki/amos/index.php?title=AMOS">AMOS</a></li>
<li><a class="reference external" href="https://github.com/najoshi/sickle">sickle</a></li>
<li><a class="reference external" href="http://picard.sourceforge.net/index.shtml">Picard</a></li>
</ul>
</div></blockquote>
<p>All programs are already installed, all you have to do is load the virtual
environment for this workshop. Once you are logged in to the server run:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">source</span> <span class="o">/</span><span class="n">proj</span><span class="o">/</span><span class="n">g2013206</span><span class="o">/</span><span class="n">metagenomics</span><span class="o">/</span><span class="n">virt_env</span><span class="o">/</span><span class="n">mg</span><span class="o">-</span><span class="n">workshop</span><span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">activate</span>
</pre></div>
</div>
<p>You deactivate the virtual environment with:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">deactivate</span>
</pre></div>
</div>
<p>NOTE: This is a python virtual environment. The binary folder of the virtual
environment has symbolic links to all programs used in this workshop so you
should be able to run those without problems.</p>
</div>
<div class="section" id="using-which-to-locate-a-program">
<h2>Using which to locate a program<a class="headerlink" href="#using-which-to-locate-a-program" title="Permalink to this headline">¶</a></h2>
<p>An easy way to determine whether you have have a certain program installed is
by typing:</p>
<div class="highlight-python"><div class="highlight"><pre>which programname
</pre></div>
</div>
<p>where <tt class="docutils literal"><span class="pre">programname</span></tt> is the name of the program you want to use. The program
<tt class="docutils literal"><span class="pre">which</span></tt> searches all directories in <tt class="docutils literal"><span class="pre">$PATH</span></tt> for the executable file
<tt class="docutils literal"><span class="pre">programname</span></tt> and returns the path of the first found hit. This is exactly
what happens when you would just type <tt class="docutils literal"><span class="pre">programname</span></tt> on the command line, but
then <tt class="docutils literal"><span class="pre">programname</span></tt> is also executed. To see what your <tt class="docutils literal"><span class="pre">$PATH</span></tt> looks like,
simply <tt class="docutils literal"><span class="pre">echo</span></tt> it:</p>
<div class="highlight-python"><div class="highlight"><pre>echo $PATH
</pre></div>
</div>
</div>
<div class="section" id="check-all-programs-in-one-go-with-which">
<h2>Check all programs in one go with which<a class="headerlink" href="#check-all-programs-in-one-go-with-which" title="Permalink to this headline">¶</a></h2>
<p>To check whether you have all programs installed in one go, you can use <tt class="docutils literal"><span class="pre">which</span></tt>.</p>
<blockquote>
<div>bowtie2
bowtie2-build
velveth
velvetg
shuffleSequences_fastq.pl
parallel
samtools</div></blockquote>
<p>Iterate over the programs calling <tt class="docutils literal"><span class="pre">which</span></tt> on each of them:</p>
<div class="highlight-python"><div class="highlight"><pre>$ req_progs=&quot;bowtie2 bowtie2-build velveth velvetg parallel samtools shuffleSequences_fastq.pl&quot;
$ echo $req_progs
bowtie2 bowtie2-build velveth velvetg parallel samtools shuffleSequences_fastq.pl
$ for p in $req_progs; do which $p || echo $p not in PATH; done
/home/idb/bin/bowtie2
/home/idb/bin/bowtie2-build
/home/idb/downloaded_software/velvet_1.2.03/velveth
/home/idb/downloaded_software/velvet_1.2.03/velvetg
/usr/local/bin/parallel
/home/idb/bin/samtools
</pre></div>
</div>
<p>As you can see all programs are found, except tablet in this case. If any of
the installed programs is missing, try to install them yourself or ask. If you
are having troubles following these examples, try to find some bash tutorials
online next time you have some time to kill. Educating yourself on how to use
the command line effectively increases your productivity immensely.</p>
</div>
<div class="section" id="optional-excercise-install-sickle-by-yourself">
<h2>(Optional excercise) Install sickle by yourself<a class="headerlink" href="#optional-excercise-install-sickle-by-yourself" title="Permalink to this headline">¶</a></h2>
<p>Follow these steps only if you want to install <tt class="docutils literal"><span class="pre">sickle</span></tt> by yourself.
Installation procedures of research software often follow the same pattern.
Download the code, <em>compile</em> it and copy the binary to a location in your
<tt class="docutils literal"><span class="pre">$PATH</span></tt>. The code for sickle is on <a class="reference external" href="https://github.com/najoshi/sickle">https://github.com/najoshi/sickle</a>. I
prefer <em>compiling</em> my programs in <tt class="docutils literal"><span class="pre">~/src</span></tt> and then copying the resulting
program to my <tt class="docutils literal"><span class="pre">~/bin</span></tt> directory, which is in my <tt class="docutils literal"><span class="pre">$PATH</span></tt>. This should get
you a long way:</p>
<div class="highlight-python"><div class="highlight"><pre>mkdir -p ~/src

# Go to the source directory and clone the sickle repository
cd ~/src
git clone https://github.com/najoshi/sickle
cd sickle

# Compile the program
make

# Create a bin directory
mkdir -p ~/bin
cp sickle ~/bin
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Checking required software</a><ul>
<li><a class="reference internal" href="#programs-used-in-this-workshop">Programs used in this workshop</a></li>
<li><a class="reference internal" href="#using-which-to-locate-a-program">Using which to locate a program</a></li>
<li><a class="reference internal" href="#check-all-programs-in-one-go-with-which">Check all programs in one go with which</a></li>
<li><a class="reference internal" href="#optional-excercise-install-sickle-by-yourself">(Optional excercise) Install sickle by yourself</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">Metagenomic Assembly Workshop</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="qtrim.html"
                        title="next chapter">Quality trimming Illumina paired-end reads</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../_sources/assembly/reqs.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="qtrim.html" title="Quality trimming Illumina paired-end reads"
             >next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Metagenomic Assembly Workshop"
             >previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" >Metagenomic Assembly Workshop</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Johannes Alneberg, Johan Bengtsson-Palme, Ino de Bruijn, Luisa Hugerth, Mikael Huss, Thomas Svensson.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.2.
    </div>
  </body>
</html>