<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Functional annotation &mdash; Metagenomics Workshop SciLifeLab 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Metagenomics Workshop SciLifeLab 1.0 documentation" href="../index.html" />
    <link rel="up" title="Comparative Functional Analysis of Metagenomic Samples" href="index.html" />
    <link rel="next" title="Determine gene coverage in metagenomic samples" href="genecoverage.html" />
    <link rel="prev" title="Gene finding" href="genefinding.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="genecoverage.html" title="Determine gene coverage in metagenomic samples"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="genefinding.html" title="Gene finding"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" accesskey="U">Comparative Functional Analysis of Metagenomic Samples</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="functional-annotation">
<h1>Functional annotation<a class="headerlink" href="#functional-annotation" title="Permalink to this headline">Â¶</a></h1>
<p>Now that we have extracted the genes/proteins we want to functionally annotate
those. There are a bunch of ways of doing this. Tomorrow you will familiarise
with BLAST searching the NCBI nr database in combination with MEGAN, which is
one possibility, but here will use one of many existing web services for
getting functional information. We will use webMGA to do do rpsBLAST searches
against the COG database. COGs are clusters of orthologs genes, i.e.
evolutionary counterparts in different species, usually with the same function
(<a class="reference external" href="http://www.ncbi.nlm.nih.gov/COG/">http://www.ncbi.nlm.nih.gov/COG/</a>). Many COGs have known functions and the COGs
are also grouped at a higher level with functional classes. To download your
newly created file open a new terminal and type:</p>
<div class="highlight-python"><pre>mkdir -p ~/metagenomics_workshop2/
scp username@kalkyl.uppmax.uu.se:~/metagenomics_workshop2/prodigal/baltic-sea-ray-noscaf-41.1000.aa.fa .</pre>
</div>
<p>To get COG classifications of your proteins, go to webMGA
<a class="reference external" href="http://weizhong-lab.ucsd.edu/metagenomic-analysis/">http://weizhong-lab.ucsd.edu/metagenomic-analysis/</a> and select Server  /
Function annotation / COG. Upload the protein file
(baltic-sea-ray-noscaf-41.1000.aa.fa) and use the default -e value cutoff.
rpsBLAST is used, which is a BLAST based on position specific scoring matrices
(pssm). For each COG, one such pssm has been constructed. These are compiled
into a database of profiles that is searched against.
<a class="reference external" href="http://www.ncbi.nlm.nih.gov/staff/tao/URLAPI/wwwblast/node20.html">http://www.ncbi.nlm.nih.gov/staff/tao/URLAPI/wwwblast/node20.html</a>. rpsBLAST is
more sensitive than a normal BLAST, which is important if genomes in your
metagenome are distant from existing sequences in databases. It is also faster
than searching against all proteins out there.</p>
<p>When the search is done you get a zipped folder. On kalkyl, create the
directory:</p>
<div class="highlight-python"><pre>mkdir -p ~/metagenomics_workshop2/wmga-cog</pre>
</div>
<p>Use wget or curl to download the zip file on uppmax or use scp to upload it to
that folder i.e.:</p>
<div class="highlight-python"><pre>scp output.zip username@kalkyl.uppmax.uu.se:~/metagenomics_workshop2/wmga-cog</pre>
</div>
<p>Then unzip the file on kalkyl:</p>
<div class="highlight-python"><pre>cd ~/metagenomics_workshop2/wmga-cog
unzip output.zip</pre>
</div>
<p>Have a look at the README.txt to see what all the files represent. The file
output.2 includes detailed information on the classifications for every protein
with a hit below the -e value cutoff. View them with:</p>
<div class="highlight-python"><pre>less README.txt
less -S output.2</pre>
</div>
<p>NOTE: If the queueing takes too much time you can also just copy the results
from the project dir:</p>
<div class="highlight-python"><pre>cp -r /proj/g2013206/metagenomics/wmga-cog/ ~/metagenomics_workshop2/</pre>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="genefinding.html"
                        title="previous chapter">Gene finding</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="genecoverage.html"
                        title="next chapter">Determine gene coverage in metagenomic samples</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../_sources/comparative-functional-analysis/annotation.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="genecoverage.html" title="Determine gene coverage in metagenomic samples"
             >next</a> |</li>
        <li class="right" >
          <a href="genefinding.html" title="Gene finding"
             >previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" >Comparative Functional Analysis of Metagenomic Samples</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Johannes Alneberg, Johan Bengtsson-Palme, Ino de Bruijn, Luisa Hugerth, Mikael Huss, Thomas Svensson.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b3.
    </div>
  </body>
</html>