<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Bonus exercise: Using Metaxa2 to investigate the taxonomic content &mdash; Metagenomics Workshop SciLifeLab 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Metagenomics Workshop SciLifeLab 1.0 documentation" href="../index.html" />
    <link rel="up" title="Metagenomic Annotation Workshop" href="index.html" />
    <link rel="prev" title="Estimating differentially abundant protein families in the metagenomes" href="differential.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="differential.html" title="Estimating differentially abundant protein families in the metagenomes"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" accesskey="U">Metagenomic Annotation Workshop</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="bonus-exercise-using-metaxa2-to-investigate-the-taxonomic-content">
<h1>Bonus exercise: Using Metaxa2 to investigate the taxonomic content<a class="headerlink" href="#bonus-exercise-using-metaxa2-to-investigate-the-taxonomic-content" title="Permalink to this headline">¶</a></h1>
<p>Now when we are familiar to using R, we can just as well use it to go through
another type of output generated by the Metaxa2 software. Metaxa2 does
classification of rRNA at different taxonomic levels, assigning each read to
a taxonomic affiliation only if it is reliably able to (given conservation
between taxa etc.) You can read more about Metaxa2 here:
<a class="reference external" href="http://microbiology.se/software/metaxa2">http://microbiology.se/software/metaxa2</a></p>
<div class="section" id="install-metaxa2">
<h2>Install Metaxa2<a class="headerlink" href="#install-metaxa2" title="Permalink to this headline">¶</a></h2>
<p>For this exercise to work, you need Metaxa2 installed. If you did not do this
earlier, here are the installation instructions again. If you did install
Metaxa2 at the beginning of the workshop, you can skip this step and move
straight to the next heading!</p>
<p>The code for Metaxa2 is available from <a class="reference external" href="http://microbiology.se/sw/Metaxa2_2.0rc3.tar.gz">http://microbiology.se/sw/Metaxa2_2.0rc3.tar.gz</a>
You can install Metaxa2 as follows:</p>
<div class="highlight-python"><div class="highlight"><pre># Create a src and a bin directory
mkdir -p ~/src
mkdir -p ~/bin

# Go to the source directory and download the Metaxa2 tarball
cd ~/src
wget http://microbiology.se/sw/Metaxa2_2.0rc3.tar.gz
tar -xzvf Metaxa2_2.0rc3.tar.gz
cd Metaxa2_2.0rc3

# Run the installation script
./install_metaxa2

# Try to run Metaxa2 (this should bring up the main options for the software)
metaxa2 -h
</pre></div>
</div>
<p>If this did not work, you can try this manual approach:</p>
<div class="highlight-python"><div class="highlight"><pre>cd ~/src/Metaxa2_2.0rc3
cp -r metaxa2* ~/bin/

# Then try to run Metaxa2 again
metaxa2 -h
</pre></div>
</div>
<p>If this brings up the help message, you are all set!</p>
</div>
<div class="section" id="generating-family-level-taxonomic-counts">
<h2>Generating family level taxonomic counts<a class="headerlink" href="#generating-family-level-taxonomic-counts" title="Permalink to this headline">¶</a></h2>
<p>If you have already run Metaxa2 to get the number of 16S rRNA sequences,
you can use the output of those runs. Otherwise you need to run the
following command on all the raw read data from all libraries:</p>
<div class="highlight-python"><div class="highlight"><pre>metaxa2 -i &lt;input file&gt; -o &lt;output file&gt; --cpu 16 --align none
</pre></div>
</div>
<p>To get counts on the family level from the metaxa2 output, we will use
another tool bundled with the Metaxa2 package; the Metaxa2 Taxonomic
Travesal Tool (<tt class="docutils literal"><span class="pre">metaxa2_ttt</span></tt>). Take a look at its options by typing:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">metaxa2_ttt</span> <span class="o">-</span><span class="n">h</span>
</pre></div>
</div>
<p>In this exercise we are interested in bacterial counts only , so we will
use the <tt class="docutils literal"><span class="pre">-t</span> <span class="pre">b</span></tt> option. Since we are only interested in family abundance
(we have too little data to get any good genus or species counts), we will
only output the data for phyla, classes, orders and families, that is we
will use the <tt class="docutils literal"><span class="pre">-m</span> <span class="pre">5</span></tt> option. As input files, you should use the files
ending with &#8221;.taxonomy.txt&#8221; that Metaxa2 produced as output. That should
give you a command looking like this:</p>
<div class="highlight-python"><div class="highlight"><pre>metaxa2_ttt -i &lt;metaxa .taxonomy.txt file&gt; -o &lt;output file&gt; -m 5 -t b
</pre></div>
</div>
<p>Run this command on the taxonomy.txt files from all input libraries. It
should be really quick. If you type <tt class="docutils literal"><span class="pre">ls</span></tt> you will notice that <tt class="docutils literal"><span class="pre">metaxa2_ttt</span></tt>
produced a bunch of .level_X.txt files. Those are the files we are going
to work with next.</p>
</div>
<div class="section" id="visualizing-family-level-taxonomic-counts">
<h2>Visualizing family level taxonomic counts<a class="headerlink" href="#visualizing-family-level-taxonomic-counts" title="Permalink to this headline">¶</a></h2>
<p>To visualize the family level counts, we will once again use R. Fire it
up again and load in the count tables from Metaxa2:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">R</span>
<span class="n">b1_fam</span> <span class="o">=</span> <span class="n">read</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="s">&quot;baltic1.level_5.txt&quot;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&quot;</span><span class="se">\t</span><span class="s">&quot;</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
<p>Repeat this procedure for all four data set. If you saved your workspace,
the merge_four function should still be available. You can try it out on the
taxonomic counts:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">all_fam</span> <span class="o">=</span> <span class="n">merge_four</span><span class="p">(</span><span class="n">b1_fam</span><span class="p">,</span><span class="n">b2_fam</span><span class="p">,</span><span class="n">swe_fam</span><span class="p">,</span><span class="n">ind_fam</span><span class="p">,</span><span class="n">c</span><span class="p">(</span><span class="s">&quot;Baltic 1&quot;</span><span class="p">,</span><span class="s">&quot;Baltic 2&quot;</span><span class="p">,</span><span class="s">&quot;Sweden&quot;</span><span class="p">,</span> <span class="s">&quot;India&quot;</span><span class="p">))</span>
</pre></div>
</div>
<p>Let&#8217;s load in the <tt class="docutils literal"><span class="pre">gplots</span></tt> library again, and make a heatmap of the raw
data:</p>
<div class="highlight-python"><div class="highlight"><pre>library(gplots)
heatmap.2(all_fam, trace = &quot;none&quot;, col = c(&quot;grey&quot;,colorpanel(255,&quot;black&quot;,&quot;red&quot;,&quot;yellow&quot;)), margin = c(5,10), cexCol = 1, cexRow = 0.7)
</pre></div>
</div>
<p>As you will notice, we will need to do some tweaking to fit in the taxonomic data:</p>
<div class="highlight-python"><div class="highlight"><pre>heatmap.2(all_fam, trace = &quot;none&quot;, col = c(&quot;grey&quot;,colorpanel(255,&quot;black&quot;,&quot;red&quot;,&quot;yellow&quot;)), margin = c(5,30), cexCol = 1, cexRow = 0.7)
</pre></div>
</div>
</div>
<div class="section" id="apply-normalizations">
<h2>Apply normalizations<a class="headerlink" href="#apply-normalizations" title="Permalink to this headline">¶</a></h2>
<p>As you might already have guessed, taxonomic count data suffers from the same
biases from, for example, sequencing library size as other gene data. To
account for that, we will apply a normalization procedure. Please note that
the normalization methods 2 and 3 (number of 16S and number of total matches
to database) would in this case be the same. In other words, the both yield
the relative abundances of the taxa. We will therefore only look at two
normalization procedures in this part of the lab.</p>
<p>First, we will normalize to the number of reads in each sequencing library.
Find the note you have taken on the data set sizes. Then apply a command like
this on the data:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">b1_fam_norm1</span> <span class="o">=</span> <span class="n">b1_fam</span> <span class="o">/</span> <span class="mi">118025</span> <span class="o">*</span> <span class="mi">1000000</span>
</pre></div>
</div>
<p>That will give you the 16S rRNA counts for the different families per million
reads. Do the same thing for the other data sets.</p>
<p>Next, we will do the same for the other type of normalization, the division
by the mapped number of reads/total number of 16S rRNA. This can, once more,
be done by dividing the vector by its sum:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">b1_fam_norm2</span> <span class="o">=</span> <span class="n">b1_fam</span> <span class="o">/</span> <span class="nb">sum</span><span class="p">(</span><span class="n">b1_fam</span><span class="p">)</span>
</pre></div>
</div>
<p>Follow the above procedure for all the data sets, and store the final
result from <tt class="docutils literal"><span class="pre">merge_four</span></tt> into a variable, for example called <tt class="docutils literal"><span class="pre">fam_norm2</span></tt>.</p>
</div>
<div class="section" id="comparing-taxonomic-distributions">
<h2>Comparing taxonomic distributions<a class="headerlink" href="#comparing-taxonomic-distributions" title="Permalink to this headline">¶</a></h2>
<p>Next we will compare the taxonomic composition of the four environments.
Let&#8217;s start out by just using a barplot. To get the different taxa on
the x-axis, we will transform the matrix with normalized counts using the
<tt class="docutils literal"><span class="pre">t()</span></tt> command. But first we need to set the margins to fit the taxonomic
names:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">par</span><span class="p">(</span><span class="n">mar</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="mi">25</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">2</span><span class="p">))</span>
<span class="n">barplot</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">fam_norm1</span><span class="p">),</span> <span class="n">main</span> <span class="o">=</span> <span class="s">&quot;Counts per million reads&quot;</span><span class="p">,</span> <span class="n">las</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">cex</span><span class="o">.</span><span class="n">names</span> <span class="o">=</span> <span class="mf">0.6</span><span class="p">,</span> <span class="n">beside</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
</pre></div>
</div>
<p>We can then do the same for the relative abundances:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">barplot</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">fam_norm2</span><span class="p">),</span> <span class="n">main</span> <span class="o">=</span> <span class="s">&quot;Relative abundance&quot;</span><span class="p">,</span> <span class="n">las</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">cex</span><span class="o">.</span><span class="n">names</span> <span class="o">=</span> <span class="mf">0.6</span><span class="p">,</span> <span class="n">beside</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
</pre></div>
</div>
<p>To only look at families present in at least two samples, we can use the
following command for filtering:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fam_norm1_filter</span> <span class="o">=</span> <span class="n">fam_norm1</span><span class="p">[</span><span class="n">rowSums</span><span class="p">(</span><span class="n">fam_norm1</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&gt;=</span> <span class="mi">2</span><span class="p">,]</span>
<span class="n">barplot</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">fam_norm1_filter</span><span class="p">),</span> <span class="n">main</span> <span class="o">=</span> <span class="s">&quot;Counts per million reads&quot;</span><span class="p">,</span> <span class="n">las</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">cex</span><span class="o">.</span><span class="n">names</span> <span class="o">=</span> <span class="mf">0.6</span><span class="p">,</span> <span class="n">beside</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Question: Which normalization method would be most suitable to use in this case? Why?</strong></p>
<p>We can also look at the differences in taxonomic content using a heatmap. As before,
we will use the squareroot as a variance stabilizing transform:</p>
<div class="highlight-python"><div class="highlight"><pre>heatmap.2(sqrt(fam_norm1), trace = &quot;none&quot;, col = c(&quot;grey&quot;,colorpanel(255,&quot;black&quot;,&quot;red&quot;,&quot;yellow&quot;)), margin = c(30,10), cexCol = 1, cexRow = 0.7)
</pre></div>
</div>
<p>Finally, we can of course also use PCA on taxonomic abundances. We will turn back to the
<tt class="docutils literal"><span class="pre">prcomp</span></tt> PCA command:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fam_norm1_pca</span> <span class="o">=</span> <span class="n">prcomp</span><span class="p">(</span><span class="n">sqrt</span><span class="p">(</span><span class="n">fam_norm1</span><span class="p">))</span>
</pre></div>
</div>
<p>We can visualize the PCA using the <tt class="docutils literal"><span class="pre">biplot</span></tt> command:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">biplot</span><span class="p">(</span><span class="n">fam_norm1_pca</span><span class="p">,</span> <span class="n">cex</span> <span class="o">=</span> <span class="mf">0.5</span><span class="p">)</span>
</pre></div>
</div>
<p>To see the proportion of variance explained by the different components, we can use the
normal plot command:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">plot</span><span class="p">(</span><span class="n">fam_norm1_pca</span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Question: Can you think about any other type of problem with the data we are using now?
This problem applies to both kinds of data, but should be particularly problematic with
the taxonomic counts...</strong></p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Bonus exercise: Using Metaxa2 to investigate the taxonomic content</a><ul>
<li><a class="reference internal" href="#install-metaxa2">Install Metaxa2</a></li>
<li><a class="reference internal" href="#generating-family-level-taxonomic-counts">Generating family level taxonomic counts</a></li>
<li><a class="reference internal" href="#visualizing-family-level-taxonomic-counts">Visualizing family level taxonomic counts</a></li>
<li><a class="reference internal" href="#apply-normalizations">Apply normalizations</a></li>
<li><a class="reference internal" href="#comparing-taxonomic-distributions">Comparing taxonomic distributions</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="differential.html"
                        title="previous chapter">Estimating differentially abundant protein families in the metagenomes</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../_sources/annotation/metaxa2.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="differential.html" title="Estimating differentially abundant protein families in the metagenomes"
             >previous</a> |</li>
        <li><a href="../index.html">Metagenomics Workshop SciLifeLab 1.0 documentation</a> &raquo;</li>
          <li><a href="index.html" >Metagenomic Annotation Workshop</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Johannes Alneberg, Johan Bengtsson-Palme, Ino de Bruijn, Luisa Hugerth, Mikael Huss, Thomas Svensson.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.2.
    </div>
  </body>
</html>